src/Action.js

```diff
@@ -1,137 +1,131 @@
+	hasFieldByName(fieldName) {
+		return hasProperty(this._fieldsByName, fieldName);
 	}
 
-	this.method = action.method || 'GET';
+	hasFieldByClass(fieldClass) {
+		return hasProperty(this._fieldsByClass, fieldClass);
+	}
+
+	hasFieldByType(fieldType) {
+		return hasProperty(this._fieldsByType, fieldType);
+	}
 
-	if (action.title) {
-		this.title = action.title;
+	getFieldByName(fieldName) {
+		return getMatchingValue(this._fieldsByName, fieldName);
 	}
 
-	this.type = action.type || 'application/x-www-form-urlencoded';
+	getFieldByClass(fieldClass) {
+		const vals = getMatchingValue(this._fieldsByClass, fieldClass);
+		return vals ? vals[0] : undefined;
+	}
 
-	this._fieldsByName = {};
-	this._fieldsByClass = {};
-	this._fieldsByType = {};
-	if (action.fields) {
-		this.fields = [];
+	getFieldsByClass(fieldClass) {
+		const vals = getMatchingValue(this._fieldsByClass, fieldClass);
+		return vals ? vals.slice() : [];
+	}
 
-		action.fields.forEach(field => {
-			const fieldInstance = new Field(field);
-			this.fields.push(fieldInstance);
+	getFieldByClasses(fieldClasses) {
+		const vals = getMatchingValuesByAll(this.fields, fieldClasses, 'class');
+		return vals && vals.length > 0 ? vals[0] : undefined;
+	}
 
-			this._fieldsByName[field.name] = fieldInstance;
+	getFieldsByClasses(fieldClasses) {
+		const vals = getMatchingValuesByAll(this.fields, fieldClasses, 'class');
+		return vals && vals.length > 0 ? vals.slice() : [];
+	}
 
-			if (fieldInstance.type) {
-				this._fieldsByType[fieldInstance.type] = this._fieldsByType[fieldInstance.type] || [];
-				this._fieldsByType[fieldInstance.type].push(fieldInstance);
-			}
+	getFieldByType(fieldType) {
+		const vals = getMatchingValue(this._fieldsByType, fieldType);
+		return vals ? vals[0] : undefined;
+	}
 
-			if (fieldInstance.class) {
-				fieldInstance.class.forEach(cls => {
-					this._fieldsByClass[cls] = this._fieldsByClass[cls] || [];
-					this._fieldsByClass[cls].push(fieldInstance);
-				});
-			}
-		});
+	getFieldsByType(fieldType) {
+		const vals = getMatchingValue(this._fieldsByType, fieldType);
+		return vals ? vals.slice() : [];
 	}
 }
-
-Action.prototype.toJSON = function() {
-	return {
-		name: this.name,
-		href: this.href,
-		class: this.class,
-		method: this.method,
-		title: this.title,
-		type: this.type,
-		fields: this.fields
-	};
-};
-
-Action.prototype.hasClass = function(cls) {
-	return this.class instanceof Array && contains(this.class, cls);
-};
-
-Action.prototype.hasField = function(fieldName) {
-	return this.hasFieldByName(fieldName);
-};
-
-Action.prototype.hasFieldByName = function(fieldName) {
-	return hasProperty(this._fieldsByName, fieldName);
-};
-
-Action.prototype.hasFieldByClass = function(fieldClass) {
-	return hasProperty(this._fieldsByClass, fieldClass);
-};
-
-Action.prototype.hasFieldByType = function(fieldType) {
-	return hasProperty(this._fieldsByType, fieldType);
-};
-
-Action.prototype.getField = function(fieldName) {
```

Comment:
These `getX` or `hasX` functions are long-outdated; I removed them from the docs in #10 (in 2017!!!). I wasn't a fan (and still am not) as they don't specify what you're getting/has-ing the `X` by.